src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -94,11 +95,67 @@ export class ElasticsearchService implements CoreService<InternalElasticsearchSe
 
     const config = await this.config$.pipe(first()).toPromise();
 
+    const adminClient$ = clients$.pipe(map(clients => clients.adminClient));
+    const dataClient$ = clients$.pipe(map(clients => clients.dataClient));
+
+    const adminClient = {
+      async callAsInternalUser(
+        endpoint: string,
+        clientParams: Record<string, any> = {},
+        options?: CallAPIOptions
+      ) {
+        const client = await adminClient$.pipe(take(1)).toPromise();
+        return await client.callAsInternalUser(endpoint, clientParams, options);
+      },
+      asScoped(request: ScopeableRequest) {
+        return {
+          callAsInternalUser: adminClient.callAsInternalUser,
+          async callAsCurrentUser(
+            endpoint: string,
+            clientParams: Record<string, any> = {},
+            options?: CallAPIOptions
+          ) {
```

Comment:
NIT: does explicitly typing the clients to `IClusterClient` avoid to redefine the `callAsXXXUser` parameter types?

Reply:
nope. TS complaints `Parameter '...' implicitly has an 'any' type`

Reply:
😞